### PR TITLE
chore: adding docs about how to add headers to grpcClient

### PR DESCRIPTION
**Pull request by franciscolopezsancho**
https://github.com/lightbend/kalix-runtime/issues/2708

### CODE REVIEW

File: samples/java-protobuf-doc-snippets/src/main/java/com/example/DelegatingServiceAction.java [hunk 1]
```diff
@@ -37,4 +38,29 @@ public Effect<DelegatingServiceApi.Result> addAndReturn(DelegatingServiceApi.Req
...
+    // tag::delegating-action-lifted[]
+    SingleResponseRequestBuilder<CounterApi.IncreaseValue, Empty> counterServiceClientBuilder
+            = ((CounterServiceClient) counterService).increase().addHeader("key", "value"); // <1>
+
+
+    CounterApi.IncreaseValue increaseValue = CounterApi.IncreaseValue.newBuilder()
+            .setCounterId(request.getCounterId())
+            .setValue(1)
+            .build();
+    CompletionStage<Empty> increaseCompleted = counterServiceClientBuilder.invoke(increaseValue);  // <2>
```
Comment (johanandren) [inline samples/java-protobuf-doc-snippets/src/main/java/com/example/DelegatingServiceAction.java:53]: I'd say the canonical use is to never catch the builder, avoiding the type soup of the builder

```
CompletionStage<Empty> increaseCompleted = 
    ((CounterServiceClient) counterService).increase()
        .addHeader("key", "value") // <1>
        .invoke(increaseValue); // <2>

File: docs/src/modules/java-protobuf/pages/call-another-service.adoc [hunk 2]
```diff
@@ -70,6 +70,34 @@ include::example$scala-protobuf-doc-snippets/src/main/scala/com/example/Delegati
...
++
+[source,scala,indent=0]
+.src/main/scala/com/example/DelegatingServiceAction.scala
+----
+include::example$scala-protobuf-doc-snippets/src/main/scala/com/example/DelegatingServiceAction.scala[tag=delegating-action-lifted]
+----
++
+<1> Casting, creating the request builder with `increase()`, and adding the headers to it.
+<2> Invoke the builder with the desired `increaseValue` parameter.
+
```
Comment (johanandren) [inline docs/src/modules/java-protobuf/pages/call-another-service.adoc:100]: Only documenting this for Java, what about Scala?

**Review by johanandren — APPROVED** [review]
LGTM (but we should maybe consider making it possible to look up a client using the client class directly)

File: samples/java-protobuf-doc-snippets/src/main/java/com/example/DelegatingServiceAction.java (hunk 1 above)
Comment (franciscolopezsancho) [reply samples/java-protobuf-doc-snippets/src/main/java/com/example/DelegatingServiceAction.java:53]: I wasn't sure. I wanted the user to know about that class. I think it's clearer in that respect, what's happening underneath. But it's true that it's a bit messy. 
I can use the canonical/cleaner way, np.

Comment (franciscolopezsancho) [reply samples/java-protobuf-doc-snippets/src/main/java/com/example/DelegatingServiceAction.java]: done

**Pull request merged**